lib/ship_fosdick/individual_shipment_updater.rb

```diff
@@ -0,0 +1,68 @@
+module ShipFosdick
+  class IndividualShipmentUpdater
+    class InvalidTransitionError < StandardError; end
+
+    def initialize(manifest_row,
+                   shipment_repository: Spree::Shipment,
+                   invalid_transition_error_factory: InvalidTransitionError)
+
+      @manifest_row = ManifestRow.new(manifest_row)
+      @shipment_repository = shipment_repository
+      @invalid_transition_error_factory = invalid_transition_error_factory
+    end
+
+    def perform
+      return unless can_update?
+
+      raise_invalid_transition_error unless shipment.can_ship?
+
+      update_attributes
```

Comment:
I am gonna say, let's have this one raise an error, and I'm wondering now if `update_order` should also raise?